### PR TITLE
Update fips tests

### PR DESCRIPTION
**Pull request by lucasmoura**
## Why is this needed?
We want to improve our integration tests to make them more resilient and easier to fix in case something goes wrong.
One way to achieve that is by consolidating as many tests as we can, so we have less places to apply a fix if needed.

We have reviewed our FIPS tests and we have consolidated some tests while also deleting some other tests that are
already covered by other scenarios.

## Test Steps
All of the modified FIPS tests should pass and we need to insure that we are not loosing anything by deleting the tests in this PR

---

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

File: features/enable_fips_vm.feature [hunk 1]
```diff
@@ -5,26 +5,46 @@ Feature: FIPS enablement in lxd VMs
...
     When I attach `contract_token` with sudo
-    When I run `pro status --format json` with sudo
-    Then stdout contains substring
+    And I apt install `jq`
+    And I run shell command `pro status --format json` with sudo
+    And I apply this jq filter `'.services[] | select(.name == "livepatch") | .available'` to the output
+    Then I will see the following on stdout
       """
-      {"available": "yes", "blocked_by": [{"name": "livepatch", "reason": "Livepatch cannot be enabled while running the official FIPS certified kernel. If you would like a FIPS compliant kernel with additional bug fixes and security updates, you can use the FIPS Updates service with Livepatch.", "reason_code": "livepatch-invalidates-fips"}], "description": "NIST-certified FIPS crypto packages", "description_override": null, "entitled": "yes", "name": "fips", "status": "disabled", "status_details": "FIPS is not configured", "warning": null}
+      "yes"
```
Comment (renanrodrigo) [inline features/enable_fips_vm.feature:13]: sorry I do think we can have a second check here for the FIPS blocked_by field. I know we may test the functionality/interaction between FIPS/Livepatch later in the flow, but here we want to assure we show the blocked_by reason in the API imho

**Review by renanrodrigo — APPROVED** [review]
LGTM way less repetitions of the same definitions.
+1 readability, +1 maintainability.
Just a small nit on the previously mentioned check

**Comment by orndorffgrant** [comment]
reposting my suggestion here (sorry again I caused so much PR churn :cry: 

This looks great!

I'd prefer we use https://pypi.org/project/jq/ over the system-under-test's `jq` binary.

I think this diff would do it:
```diff
diff --git a/features/enable_fips_vm.feature b/features/enable_fips_vm.feature
index d36d1d87..7d6d9236 100644
--- a/features/enable_fips_vm.feature
+++ b/features/enable_fips_vm.feature
@@ -7,7 +7,7 @@ Feature: FIPS enablement in lxd VMs
     When I attach `contract_token` with sudo
-    And I apt install `jq`
     And I run shell command `pro status --format json` with sudo
-    And I apply this jq filter `'.services[] | select(.name == "livepatch") | .available'` to the output
+    And I apply this jq filter `.services[] | select(.name == "livepatch") | .available` to the output
     Then I will see the following on stdout
       """
       "yes"
@@ -36,13 +36,13 @@ Feature: FIPS enablement in lxd VMs
     And I ensure apt update runs without errors
     And I verify that `<fips-packages>` are installed from apt source `<fips-apt-source>`
     When I run shell command `pro status --format json --all` with sudo
-    And I apply this jq filter `'.services[] | select(.name == "livepatch") | .available'` to the output
+    And I apply this jq filter `.services[] | select(.name == "livepatch") | .available` to the output
     Then I will see the following on stdout
       """
       "no"
       """
     When I run shell command `pro status --format json --all` with sudo
-    And I apply this jq filter `'.services[] | select(.name == "livepatch") | .blocked_by[0].reason'` to the output
+    And I apply this jq filter `.services[] | select(.name == "livepatch") | .blocked_by[0].reason` to the output
     Then I will see the following on stdout
       """
       "Livepatch cannot be enabled while running the official FIPS certified kernel. If you would like a FIPS compliant kernel with additional bug fixes and security updates, you can use the FIPS Updates service with Livepatch."
diff --git a/features/steps/output.py b/features/steps/output.py
index 673b54db..c029bc56 100644
--- a/features/steps/output.py
+++ b/features/steps/output.py
@@ -4,6 +4,7 @@ import re
 import textwrap
 
 import jsonschema  # type: ignore
+import jq
 import yaml
 from behave import then, when
 from hamcrest import assert_that, contains_string, equal_to, not_
@@ -249,16 +250,4 @@ def i_verify_field_is_redacted_in_the_logs(context, field):
 
 @when("I apply this jq filter `{jq_filter}` to the output")
 def i_apply_jq_filter(context, jq_filter):
-    file_path = "/tmp/json-output"
-    when_i_create_file_with_content(
-        context, file_path, text=context.process.stdout.strip()
-    )
-    cmd = "cat {} | jq {}".format(
-        file_path,
-        jq_filter.replace('"', '\\"'),
-    )
-    when_i_run_shell_command(
-        context,
-        cmd,
-        "with sudo",
-    )
+    context.process.stdout = jq.compile(jq_filter).input_text(context.process.stdout.strip()).text()
diff --git a/tox.ini b/tox.ini
index 3b7308fe..284e4bd7 100644
--- a/tox.ini
+++ b/tox.ini
@@ -56,6 +56,7 @@ commands: pylint {posargs:uaclient lib apport}
 deps =
     behave
     jsonschema
+    jq
     pycloudlib==1!6.5.0
     PyHamcrest
     requests
```

Also, do you think we can combine all the `enable_fips_*.feature` files into one `fips.feature` file? Or would that be too big?

File: features/enable_fips_vm.feature [hunk 2]
```diff
@@ -5,26 +5,45 @@ Feature: FIPS enablement in lxd VMs
...
+      Updating standard Ubuntu package lists
+      Installing FIPS packages
+      FIPS enabled
+      A reboot is required to complete install.
+      """
     Then stdout matches regexp:
       """
       This will install the FIPS packages. The Livepatch service will be unavailable.
       Warning: This action can take some time and cannot be undone.
       """
```
[the remark this answers is not in the document]
Comment (lucasmoura) [reply features/enable_fips_vm.feature:40]: The problem is that on focal, we have the kvm kernel installed by default, which trigger this extra message here

**Review by orndorffgrant — APPROVED** [review]
Some minor comments so preemptively approving

File: features/enable_fips_vm.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (orndorffgrant) [reply features/enable_fips_vm.feature:40]: I think all lxd vms prior to noble use the kvm kernel - but I'd need to verify that.
The regexp check immediately after this new one will still pass even with the extra message though I think

File: features/enable_fips_vm.feature [hunk 3]
```diff
@@ -81,77 +102,68 @@ Feature: FIPS enablement in lxd VMs
...
     And stdout contains substring:
       """
-      Updating <fips-name> package lists
+      Updating FIPS Updates package lists
       Updating standard Ubuntu package lists
-      Installing <fips-name> packages
-      <fips-name> enabled
+      Installing FIPS Updates packages
+      """
+    And if `<release>` in `jammy` and stdout matches regexp:
```
[the remark this answers is not in the document]
Comment (lucasmoura) [reply features/enable_fips_vm.feature]: Indeed, I have changed to a substring check

**Pull request merged**